python/lsst/metaserv/schemaToMeta.py

```diff
@@ -0,0 +1,271 @@
+                            in_col["defaultValue"] = dv
+                        colNum += 1
+                        if "columns" not in in_table:
+                            in_table["columns"] = []
+                        in_table["columns"].append(in_col)
+                    # print "found col: ", in_col
+                elif self._isCommentLine(line): # handle comments
+                    if in_col is None:    # table comment
+
+                        if self._containsDescrTagStart(line):
+                            if self._containsDescrTagEnd(line):
+                                in_table["description"] = self._retrDescr(line)
+                            else:
+                                in_table["description"] = self._retrDescrStart(line)
+                        elif "description" in in_table:
+                            if self._containsDescrTagEnd(line):
+                                in_table["description"] += self._retrDescrEnd(line)
+                            else:
+                                in_table["description"] += self._retrDescrMid(line)
+                    else:
+                                          # column comment
+                        if self._containsDescrTagStart(line):
+                            if self._containsDescrTagEnd(line):
+                                in_col["description"] = self._retrDescr(line)
+                            else:
+                                in_col["description"] = self._retrDescrStart(line)
+                                in_colDescr = 1
+                        elif in_colDescr:
+                            if self._containsDescrTagEnd(line):
+                                in_col["description"] += self._retrDescrEnd(line)
+                                in_colDescr = None
+                            else:
+                                in_col["description"] += self._retrDescrMid(line)
+
+                                          # units
+                        if self._isUnitLine(line):
+                            in_col["unit"] = self._retrUnit(line)
+
+                                          # ucds
+                        if self._isUcdLine(line):
+                            in_col["ucd"] = self._retrUcd(line)
+
+        iF.close()
+        return table
+
+    ###########################################################################
+    # Helper functions
+    ###########################################################################
+
+    def _isIndexDefinition(self, c):
+        return c in ["PRIMARY", "KEY", "INDEX", "UNIQUE"]
+
+    def _isCommentLine(self, str):
+        return re.match('[\s]*--', str) is not None
+
+    def _isUnitLine(self, str):
+        return re.search(r'<unit>(.+)</unit>', str) is not None
+
+    def _isUcdLine(self, str):
+        return re.search(r'<ucd>(.+)</ucd>', str) is not None
+
+    def _retrUnit(self, str):
+        xx = re.compile(r'<unit>(.+)</unit>')
```

Comment:
Here and in other places where you do `re.search` and `re.compile` regular expression is compiled on every call to the method, which is likely a significant overhead for this call.